### DISCUSSION TITLE
player: encapsulating CSS

### DISCUSSION THREAD

**Discussion by guludo** [asciinema/discussions#159] [Ideas]
Hi there.

Currently, for self-hosting `asciinema-player`, we are supposed to include the javascript and css file using the `<script>` and `<link>` tags, respectively. While that works, I believe that the current approach can lead to conflicts with other libraries (or even the user code) regarding CSS class names. I think that could be avoided using a [shadow root](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM).

Best regards,
Gustavo Sousa

**Comment by ku1ik** [comment]
That's certainly an idea worth checking out. The major version of the player is a rewrite in Solid.js (`next` branch), I think it may support shadow root, one way or another.

**Comment by ku1ik** [comment]
Meanwhile in https://github.com/asciinema/asciinema-player/commit/90ae4a0af4077843f3270890d29714491df360b1 and https://github.com/asciinema/asciinema-player/commit/dbc10e577e48db61dcaa726291796c6e75f55e58 I updated most class names to use `ap-` prefix (ap == asciinema player), which while not 100% immune to collisions it should lower the risk.
